tests/remove_routes_issue.cc

```diff
@@ -0,0 +1,145 @@
+#include <gtest/gtest.h>
+#include <pistache/endpoint.h>
+#include <pistache/http.h>
+#include <pistache/peer.h>
+#include <pistache/router.h>
+#include <httplib.h>
+#include "rapidjson/document.h"
+#include <csignal>
+#include <iostream>
+#include <memory>
+
+using namespace Pistache;
+using namespace Rest;
+
+static constexpr auto URL_1_MODE_1 = "/read/hello_fun_1_mode_1";
+static constexpr auto URL_2_MODE_1 = "/read/hello_fun_2_mode_1";
+
+static constexpr auto URL_1_MODE_2 = "/read/hello_fun_1_mode_2";
+static constexpr auto URL_2_MODE_2 = "/read/hello_fun_2_mode_2";
+
+class Server
+{
+public:
+    enum Modes {
+        mode_1,
+        mode_2
+    };
+    Server(Address addr) :endpoint_(std::make_shared<Pistache::Http::Endpoint>(addr))
+    {}
+    void init(size_t thr = 2, Modes mode = Modes::mode_1){
+        auto opts = Pistache::Http::Endpoint::options()
+                        .threads(static_cast<int>(thr));
+        endpoint_->init(opts);
+        system_mode_ = mode;
+        SetupRoutes();
+    }
+    virtual ~Server() = default;
+    void SwitchMode()
+    {
+        if (system_mode_ == Modes::mode_1)
+        {
+            std::cout << "Setup remove route mode_1" <<  std::endl;
+            Routes::Remove(router_, Pistache::Http::Method::Get, URL_1_MODE_1);
+            Routes::Remove(router_, Pistache::Http::Method::Get, URL_2_MODE_1);
+            system_mode_ = Modes::mode_2;      
+        }
+        else
+        {
+            std::cout << "Setup remove route mode_2" <<  std::endl;
+            Routes::Remove(router_, Pistache::Http::Method::Get, URL_1_MODE_2);
+            Routes::Remove(router_, Pistache::Http::Method::Get, URL_2_MODE_2);
+            system_mode_ = Modes::mode_1;
+        }
+        SetupRoutes();
+    }
+    void StartServer()
+    {
+        endpoint_->setHandler(router_.handler());
+        endpoint_->serveThreaded();
+    }
+    void StopServer()
+    {
+        endpoint_->shutdown();
+    }
+    Port getPort()
+    {
+        return endpoint_->getPort();
+
+    }
+
+private:
+    Router router_;
```

Comment:
use a shared pointer here is mandatory if we want to make the route table mutable, with this implementation the route is not mutable that explain why   PR fails